### ISSUE TITLE
[HL2] Muzzle flashes render issue

### ISSUE THREAD

**Issue by VladimirVSC** [ValveSoftware/Source-1-Games#3778]
The muzzle flashes are either missing or rendered just partially (half or even a quarter of the effect). The most interresting is that they look normal, when on screenshots. Maybe it somehow connected with my monitor (it has a fake 100hz feature with real 60hz frequency)? If so, can I do something with it without changing the monitor.

System: Windows 7 SP1, GTX 760 2Gb, CoreI5-4430

**Comment by shiro-2k** [comment]
@VladimirVSC Have you determined whether the issue is present with the 100hz feature temporarily turned off?

**Comment by VladimirVSC** [comment]
> @VladimirVSC Have you determined whether the issue is present with the 100hz feature temporarily turned off?

It can't be turned off. Therer is no such an option, unfortunately

**Comment by shiro-2k** [comment]
> > @VladimirVSC Have you determined whether the issue is present with the 100hz feature temporarily turned off?
> 
> It can't be turned off. Therer is no such an option, unfortunately

What model is the monitor? Also, are you able to record video of the issue by chance?

**Comment by VladimirVSC** [comment]
> > > @VladimirVSC Have you determined whether the issue is present with the 100hz feature temporarily turned off?
> > 
> > 
> > It can't be turned off. Therer is no such an option, unfortunately
> 
> What model is the monitor? Also, are you able to record video of the issue by chance?

https://user-images.githubusercontent.com/96921167/149184167-72edc55c-4960-447b-8ba2-bd33b3195226.mp4

The monitor is Samsung UE28J4100AK

**Comment by shiro-2k** [comment]
> > > > @VladimirVSC Have you determined whether the issue is present with the 100hz feature temporarily turned off?
> > > 
> > > 
> > > It can't be turned off. Therer is no such an option, unfortunately
> > 
> > 
> > What model is the monitor? Also, are you able to record video of the issue by chance?
> 
>  Half-Life.2.2022-01-12.19-29-48-50.mp4 
> The monitor is Samsung UE28J4100AK

Thanks for sharing the requested information. While testing, I noticed that I had the same problem when using the shotgun. I resolved the issue by making sure that I had selected "Use the 3D application setting" for the Vertical sync feature within the NVIDIA Control Panel, and enabling "Wait for vertical sync" in Half-Life 2's video options menu. Give this a try and let me know how it goes. My conclusion here is that the muzzle flash sprites appear missing at times due to [screen-tearing.](https://en.wikipedia.org/wiki/Screen_tearing)

**Comment by VladimirVSC** [comment]
> > Replying to [#3778 (comment)](https://github.com/ValveSoftware/Source-1-Games/issues/3778#issuecomment-1011247430)
> 
> Thanks for sharing the requested information. While testing, I noticed that I had the same problem when using the shotgun. I resolved the issue by making sure that I had selected "Use the 3D application setting" for the Vertical sync feature within the NVIDIA Control Panel, and enabling "Wait for vertical sync" in Half-Life 2's video options menu. Give this a try and let me know how it goes. My conclusion here is that the muzzle flash sprites appear missing at times due to [screen-tearing.](https://en.wikipedia.org/wiki/Screen_tearing)

You are right. Enabling the Vsync option fixed the issue completely! But Vsync enabling must burn in hell due to its input lag.
